Mathlib/Data/Finsupp/Basic.lean

```diff
@@ -1691,26 +1691,91 @@ instance uniqueOfLeft [IsEmpty α] : Unique (α →₀ R) :=
 
 end
 
+section
+variable {M : Type*} [Zero M] {P : α → Prop} [DecidablePred P]
+
+/-- Combine finitely supported functions over `{a // P a}` and `{a // ¬P a}`, by case-splitting on
+`P a`. -/
+@[simps]
+def piecewise (f : Subtype P →₀ M) (g : {a // ¬ P a} →₀ M) : α →₀ M where
```

Comment:
Whould it be possible to write `(g : Subtype (¬ P))` ?

Reply:
No. I could write `Subtype Pᶜ`, but that's a bit obtuse.

Reply:
Pᶜ is defeq abuse but (¬ P \.) should work. I would write {a // P a} though.

Reply:
> Pᶜ is defeq abuse

No it isn't, this uses the complete lattice structure on Pi and Prop.

Reply:
Either solution looks fine to me. It's just that mathlib code also serves as examples of how to write code in proper “mathlib style”, and the dissymmetry puzzled me.

Reply:
> No it isn't, this uses the boolean algebra structure on Pi and Prop.

I see. Yeah I should know that ᶜ doesn't just work for Set.

Reply:
The only reason I avoided 
```suggestion
def piecewise (f : {a // P a} →₀ M) (g : {a // ¬ P a} →₀ M) : α →₀ M where
```
is that it introduces an unnecessary eta expansion in the type